### DISCUSSION TITLE
BPL_* flag show configurated value at building log but not taking effect when run the image.

### DISCUSSION THREAD

**Discussion by skylartd** [paketo-buildpacks/feedback#202] [Java Team]
Hi Java Team,

In the java pack [documentation,](https://github.com/paketo-buildpacks/bellsoft-liberica#configuration:~:text=publish%205000%3A5000-,%24BPL_DEBUG_ENABLED,-Configure%20whether%20remote) there are environment variables prefix with **BPL_**. But when I set this value during the pack building process (either via spring boot plugin configuration or pack command). The building log showing the configurated value, but it will not take effect when we run the built image.

I can however use the [environment variables build pack](https://github.com/paketo-buildpacks/environment-variables) to set these values and it will take effect when we run the image.

Is this the expected behavior? Since the java pack log is showing the launch configuration but it will not take effect when we run the image.

Example:
<img width="824" alt="image" src="https://github.com/paketo-buildpacks/feedback/assets/20160103/07d16055-2d62-4d2c-8f5c-b45cfb501768">
The BPL_DEBUG_ENABLED has been set to true but will not do anything when we run the image.
But if we set BPE_BPL_DEBUG_ENABLED=true, it will take effect.
![image](https://github.com/paketo-buildpacks/feedback/assets/20160103/233e73ba-0d8d-4ee0-9e21-54d9e86fd099)

**Comment by dmikusa** [comment]
> Is this the expected behavior?

Yes.

Launch Configuration needs to be configured at Launch or Runtime. That is when you actually execute the image generated by buildpacks. It does not apply at build time.

The convention followed by Paketo buildpacks is that variables starting with `BP_` are build time configuration and `BPL_` are launch time configuration.

If you want to set a launch time configuration variable at build time, you can do that with the environment variable buildpack like you mentioned. That will set the value you specify at build time in a way that it will be automatically applied when you run the application image (i.e. at launch time). Values set with the environment variables buildpack work like "default" values because a user can still override this at launch time, but it's optional and the user doesn't need to set it at launch time if the user accepts the default value you set.

Hope that helps!

**Reply by skylartd** [reply]
Thank you for the clear explanation.
I was confused that it's showing the configurated value and did not take effect. But I got it now.
